skunkworks_crow/src/main/java/org/odk/share/views/ui/bluetooth/BtReceiverActivity.java

```diff
@@ -96,6 +98,7 @@ protected void onCreate(Bundle savedInstanceState) {
         // checking for if bluetooth enabled
         if (!BluetoothUtils.isBluetoothEnabled()) {
             BluetoothUtils.enableBluetooth();
+            Toast.makeText(this, getString(R.string.turning_on_bluetooth_message), Toast.LENGTH_SHORT).show();
             BtReceiverActivityPermissionsDispatcher.updateDeviceListWithPermissionCheck(this);
```

Comment:
Ideally Bluetooth should not be triggered if location permission is not there.

Reply:
Thanks a lot for checking this! I found some more places that I can improve, you can refer to the updates.

Reply:
We have already call update devices list method in `initEvents()` method, so we don't have to call that again in `onCreate()`, and I moved the enable bluetooth method into the method required location permission, now it may looks fine.